gearmand/datadog_checks/gearmand/gearmand.py

```diff
@@ -115,6 +115,17 @@ def check(self, instance):
                 message="Connection to %s:%s succeeded." % (host, port),
                 tags=tags,
             )
+            self._collect_metadata(client)
```

Comment:
This could be used before collection: https://github.com/DataDog/integrations-core/pull/5748/files

Reply:
Hmm, that method doesn't seem to support a test env very well.  I can investigate fixing it in a separate PR.

Reply:
Ah ok

Reply:
Why does it not support a test env?

Reply:
It checks the config of the `datadog_agent`, but in the test stub version that method returns a hardcoded `''` (empty string).

See here: https://github.com/DataDog/integrations-core/blob/91cb07a59f34f9879fadc5504c1362646aa5e3bb/datadog_checks_base/datadog_checks/base/stubs/datadog_agent.py#L27

Reply:
Ah I see, I made a unit test for `_submit_metadata` for the other integration I used it in so I didn't run into the issue